### PR TITLE
add generic host_port parsing based on scheme and use it in qps driver

### PR DESCRIPTION
**Pull request by apolcyn**
Provides generic versions of `gpr_split_host_port` and `gpr_join_host_port`.
These generic versions default to `gpr_split_host_port` and `gpr_join_host_port`, but this allows registering a different `host_port_parser` to be used instead for certain schemes.

### CODE REVIEW

File: src/core/ext/client_channel/generic_uri_parse.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ */
+
+#ifndef GRPC_CORE_EXT_CLIENT_CHANNEL_GENERIC_URI_PARSE_H
+#define GRPC_CORE_EXT_CLIENT_CHANNEL_GENERIC_URI_PARSE_H
+
+#include "src/core/ext/client_channel/client_channel_factory.h"
+#include "src/core/ext/client_channel/resolver.h"
+#include "src/core/ext/client_channel/uri_parser.h"
+
+typedef struct grpc_host_port_parser grpc_host_port_parser;
```
Comment (ctiller) [inline src/core/ext/client_channel/generic_uri_parse.h]: Could we eliminate this file by adding the host/port parsing as part of grpc_resolver_factory?

I find it difficult to imagine a time when I'd want to declare a grpc_resolver_factory without a corresponding grpc_host_port_parser: they seem to occupy the same semantic space.

Perhaps s/grpc_resolver_factory/grpc_naming_scheme/g?

**Comment by markdroth** [comment]
Based on the email discussion, I suspect this PR isn't needed anymore.

**Comment by apolcyn** [comment]
FTR going further into this, I realized that this as an alternative to https://github.com/grpc/grpc/pull/9168 still runs into issues when e.g., the benchmark driver is working with go workers - I don't think I'd be able to use this for the driver, should this still be continued?

**Comment by markdroth** [comment]
I'm going to close this, since it doesn't seem to be needed.  Please re-open if there's some reason we need to resurrect it.

**Pull request closed without merging**